tests/testthat/test-join_abcd_scenario.R

```diff
@@ -259,3 +259,52 @@ test_that("outputs the same with upper/lower abcd$sector or abcd$technology", {
   out_upper <- join_abcd_scenario(matched, upper_technology, scenario, regions)
   expect_equal(out_upper, out_lower)
 })
+
+test_that("outputs full timeline of scenario #157", {
+
+  out <- join_abcd_scenario(
+    fake_matched(),
+    fake_abcd(year = 2020),
+    fake_scenario(scenario = "1.5c-scen", year = c(2020, 2025)),
+    region_isos = region_isos_stable
+  )
+
+  expect_equal(max(out$year), 2025L)
+
+})
+
+test_that("doesnt output sectors that aren't in input data #157", {
+
+  out <- join_abcd_scenario(
+    fake_matched(sector_abcd = "power"),
+    fake_abcd(sector = "power", technology = "hydrocap"),
+    fake_scenario(
+      sector = c("power", "automotive"),
+      technology = c("hydrocap", "ice")
+      ),
+    region_isos = region_isos_stable
+  )
+
+  expect_equal(unique(out$sector_abcd), "power")
+
+})
+
+test_that("on fill in production timeline after t0 #157", {
```

Comment:
I don't understand what this expectation means. Looking at the test more closely it seems like the idea is to fill production only after an initial positive value, not before one. I am not sure I methodologically agree

Reply:
Firstly, noted this test is named horribly. I will go over all of the unit test names and try to make the more clear 😂 

To the methodological comment, the goal, as per George's request, is to "extend the timeline of the `target_*` outputs of the functions". 

Since target calculation only requires the production value at some t0 (either technology- or sector- total), this is doable for any arbitrary scenario timeline. 

We don't "fill in" the `production` at all.
Notice that, in the output of the function, for any years greater than what we have data for in the ABCD, the value of `production` is `NA`. 

So for example, the expectations for the output of `target_market_share` would be:
IF:
* `abcd` has production values for `2020:2025`
* `scenario` has target values for `2015:2030`
THEN:
* `out` for `metric = production` will have values between years `2020:2025`
* `out` for `metric = target_*` will have values between years `2020:2030`
* `out` will have no values prior to `2020` (the "start" year of production)

I think maybe this deserves a call to discuss, as I agree that the change is pretty drastic.

Reply:
https://github.com/RMI-PACTA/r2dii.analysis/issues/482